File: data-structures/ds3-stack/simple-array/java/SimpleArray.java
public class SimpleArrayStack {
    private int maxSize;
    private int top;
    private int[] stackArray;

    public SimpleArrayStack(int size) {
        this.maxSize = size;
        this.stackArray = new int[maxSize];
        this.top = -1;
    }

    public void push(int value) throws StackOverflowError {
    }

    public Object pop() throws RuntimeException {
        return null;
    }

    public Object peek() {
        return null;

    }

    public Object getSize() {
        return null;
    
    }

    public Boolean isEmpty() {
        return null;
    }
}

File: data-structures/ds3-stack/simple-array/javascript/simpleArray.ts
export class SimpleStackArray<T> {
  private stack: T[];

  constructor() {
    this.stack = [];
  }

  /**
   * Push an element onto the stack
   *
   * @param element The element to be pushed onto the stack
   * @returns None
   */
  push(element: T): void {}

  /**
   * Pop an element from the stack
   *
   * @returns The popped element, or undefined if the stack is empty
   */
  pop(): T | undefined {
    return;
  }

  /**
   * Peek at the top element of the stack without removing it
   *
   * @returns The top element, or undefined if the stack is empty
   */
  peek(): T | undefined {
    return;
  }

  /**
   * Check if the stack is empty
   *
   * @returns True if the stack is empty, false otherwise
   */
  isEmpty(): boolean {
    return;
  }

  /**
   * Clear the stack
   *
   * @returns None
   */
  size(): number {
    return;
  }
}
